airbyte-integrations/connectors/source-serpstat/source_serpstat/spec.yaml

```diff
@@ -0,0 +1,87 @@
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  type: object
+  required:
+    - api_key
+  properties:
+    api_key:
+      type: string
+      title: API Key
+      airbyte_secret: true
+      order: 0
+      description: >-
+        Serpstat API key can be found here:
+        https://serpstat.com/users/profile/
+    domain:
+      type: string
+      order: 1
+      title: Domain
+      default: serpstat.com
+      description: The domain name to get data for (ex. serpstat.com)
+    page_size:
+      type: integer
+      order: 2
+      title: Page size
+      default: 10
+      description: >-
+        The number of data rows per page to be returned. Each data row can
+        contain multiple data points. The max value is 1000. Reducing the size
+        of the page will result in fewer API credits spent.
+    domains:
+      type: array
+      order: 3
+      title: Domains
+      description: >-
+        The list of domains that will be used in streams that support batch
+        operations
+    filter_by:
+      type: string
+      order: 4
+      title: Filter by
+      description: >-
+        The field name by which the results should be filtered. Filtering the
+        results will result in fewer API credits spent. Each stream has
+        different filtering options. See https://serpstat.com/api/ for more
+        details.
+    filter_value:
+      type: string
+      order: 5
+      title: Filter value
+      description: >-
+        The value of the field to filter by. Each stream has different
+        filtering options. See https://serpstat.com/api/ for more details.
+    sort_by:
+      type: string
+      order: 6
+      title: Sort by
+      description: >-
+        The field name by which the results should be sorted. Each stream has
+        different sorting options. See https://serpstat.com/api/ for more
+        details.
+    sort_value:
+      type: string
+      order: 7
+      title: Sort value
+      description: >-
+        The value of the field to sort by. Each stream has different sorting
+        options. See https://serpstat.com/api/ for more details.
+    pages_to_fetch:
+      type: integer
+      order: 8
+      title: Pages to fetch
+      default: 1
+      description: >-
+        The number of pages that should be fetched. All results will be
+        obtained if left blank. Reducing the number of pages will result in

```

Comment:
Connector must ingest all data available and not only a subset of the data.

Reply:
Hi @marcosmarxm 

The reason I used the subset of data and limited page size is that our API is charged based on the number of rows returned. We can return all the data but we see a lot of cases where our users want to limit the API rows consumption. So it's a measure to prevent shooting your own leg. Unlimited values by default might eat a lot of API rows which will not make our users happier. 

Can we proceed without these requirements or at least use some default values?

Reply:
Yes, but please update the description and the documentation of the connector explaining the case.